crates/engine/src/types.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Off-chain environment account ID type.
+pub type OffAccountId = Vec<u8>;
```

Comment:
I think type alias are the worst of both worlds. They are not strongly typed, yet hide their real type behind a name that you have to memorize. Also their usage is optional which makes everything even worse. I'd definitely either use the type `Vec<u8>` (or better `[u8; 32]`) directly or make a proper strongly typed wrapper around it with some `from_bytes` and `to_bytes` methods.

Reply:
Note that for types such as `Balance` we will need `Vec<u8>` as base because they are generic over the environment.

Reply:
Forget what I said about `[u8; 32]` for `AccountId`. I confused it with `Key`. Sorry!